src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -19389,6 +19389,95 @@ public class CClass<TClass> where TClass : class
                 );
         }
 
+        [Fact]
+        public void MaybeNull_Field()
+        {
+            // Warn on misused nullability attributes (f2, f3, f4, f5)? https://github.com/dotnet/roslyn/issues/36073
+            var lib_cs =
+@"using System.Diagnostics.CodeAnalysis;
+public class COpen<TOpen>
+{
+    [MaybeNull]public TOpen f1 = default!;
+}
+public class CClass<TClass> where TClass : class
+{
+    [MaybeNull]public TClass f2 = null!;
```

Comment:
Out of scope, but I wonder how common it will be for people to misuse the attributes like this.

I think any usage of [MaybeNull] on a non-annotated reference type is probably wrong. The correct attribute usage would probably be to annotate as nullable, right?

Reply:
In this specific case, it seems okay to use the [MaybeNull]. That indicates that you could get a `null` out, but would be disallowed to put a `null` in.
I don't think that's very useful, but it is not strictly equivalent to saying `TClass? f2 = null;`